### PR TITLE
Add ADR for elasticache

### PR DESCRIPTION
**Pull request by surminus**
https://trello.com/c/VNID5ncy/676-adr-for-using-elasticache

### CODE REVIEW

File: doc/architecture/decisions/0025-use-elasticache-for-redis.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+We should consider provisioning Elasticache to replace our provisioned Redis instances.
+
+## Decision
+
+We are using Elasticache instead of provisioning our own Redis instances.
+
+## Consequences
+
+We have less Puppet code to manage. There is no risk of lock-in because the client side
```
Comment (SamLR) [inline doc/architecture/decisions/0025-use-elasticache-for-redis.md]: Are there any known limitations on the AWS version of Elasticache? E.g. config options we have less control over, lack of an actual super user (such as having to use `rds_superuser` for RDS which isn't an actual superuser).

Comment (surminus) [reply doc/architecture/decisions/0025-use-elasticache-for-redis.md]: I'll defer to @deanwilson and @afda16 who set this up. I don't have enough experience with Elasticache.

Comment (deanwilson) [reply doc/architecture/decisions/0025-use-elasticache-for-redis.md]: Yes,  there are some options we can't use under Elasticache. Mostly these are things we wouldn't want in an AWS environment but if we hit any that are actual issues for us we can amend the ADR and move back to a server/container.

Comment (SamLR) [reply doc/architecture/decisions/0025-use-elasticache-for-redis.md]: Thought so, might be worth making a note of this in the ADR? (sorry @surminus)

Comment (surminus) [reply doc/architecture/decisions/0025-use-elasticache-for-redis.md]: This has been updated.

**Pull request merged**